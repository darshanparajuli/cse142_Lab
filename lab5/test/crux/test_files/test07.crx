// test comparison

func compareInt(a:int, b:int) : bool {
  return a > b;
}

func compareFloat(a:float, b:float) : bool {
  return a > b;
}

func compareBool(a:bool, b:bool) : bool {
  return a > b;
}

func compareIntFloat(a:int, b:float) : bool {
  return a > b;
}

func main() : void {
  ::printBool(false);
}
