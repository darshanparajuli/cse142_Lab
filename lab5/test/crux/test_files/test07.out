Error type-checking file.
TypeError(12,12)[Cannot compare bool with bool.]
TypeError(12,3)[Function compareBool returns bool not ErrorType(Cannot compare bool with bool.).]
TypeError(16,12)[Cannot compare int with float.]
TypeError(16,3)[Function compareIntFloat returns bool not ErrorType(Cannot compare int with float.).]

